src/understand-tidb/plugin.md

```diff
@@ -1 +1,63 @@
 # Plugin
+
+The plugin API allows TiDB to be extended with new features such as audit logging or IP allow/deny listing.
+
+Sample code is provided for a basic audit logging plugin at [`./tidb/plugin/conn_ip_example/`](https://github.com/pingcap/tidb/tree/master/plugin/conn_ip_example). For an example on compiling TiDB and this plugin:
+
+```bash
+plugin="conn_ip_example"
+cd cmd/pluginpkg
+go install
+cd ../../plugin/$plugin
+pluginpkg -pkg-dir . -out-dir .
+cd ../..
+ 
+./bin/tidb-server -plugin-dir /path/to/tidb/plugin/$plugin -plugin-load $plugin-1
+```
+
+An explanation of what this does:
+
+- `cd cmd/pluginpkg` and `go install` compiles the command line utility called `pluginpkg`, which is used to build the plugin.
+- `pluginpkg -pkg-dir . -out-dir .` reads the plugin code + `manifest.toml` file and generates a shared object file for the plugin (`conn_ip_example-1.so`).
+- When the tidb-server starts, it can load plugins in a specified directory (`plugin-dir`).
+
```

Comment:
we can provide SQL statements or logs to verify that the plugin has been loaded.

Reply:
I've incorporated this, thanks!